runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -407,9 +411,31 @@ void J9::RecognizedCallTransformer::transform(TR::TreeTop* treetop)
       case TR::java_lang_Integer_rotateLeft:
          processIntrinsicFunction(treetop, node, TR::irol);
          break;
+      case TR::java_lang_Integer_rotateRight:
+         {
+         // rotateRight(x, distance) = rotateLeft(x, -distance)
+         TR::Node *distance = TR::Node::create(node, TR::ineg, 1);
```

Comment:
Do we know what happens for `Integer.MIN_INT` if that is an acceptable input?

Reply:
This transformation yields the correct result for `distance = INT_MIN`. All the rotates here (both the recognized methods in the JCL and the `lrol`/`irol` instructions) take their second operand modulo the number of bits in the data type being rotated as the distance by which to rotate. For an `INT_MIN` distance, this would result in a rotate by a distance of 0 for both ints and longs, so `rotateRight(x, INT_MIN) = x` and `rotateLeft(x, INT_MIN) = x`. Since `-INT_MIN = INT_MIN` when using `ineg` in our IL, it follows that `rotateRight(x, INT_MIN) = x = rotateLeft(x, INT_MIN) = rotateLeft(x, -INT_MIN)`.

Reply:
Thanks for the excellent explanation.